packages/platform/api/src/auth/auth.controller.ts

```diff
@@ -56,23 +127,190 @@ export class AuthController {
   @ApiBearerAuth()
   @ApiOperation({ summary: 'Create API key' })
   @ApiResponse({ status: 201, description: 'API key successfully created' })
-  @ApiResponse({ status: 401, description: 'Unauthorized' })
   @UseGuards(JwtAuthGuard)
   @Post('api-keys')
-  async createApiKey(
-    @Req() req: any,
-    @Body() createApiKeyDto: CreateApiKeyDto,
-  ) {
+  async createApiKey(@Req() req: any, @Body() createApiKeyDto: CreateApiKeyDto) {
     return this.authService.createApiKey(req.user.id, createApiKeyDto.name);
   }
 
   @ApiBearerAuth()
   @ApiOperation({ summary: 'Get current user profile' })
   @ApiResponse({ status: 200, description: 'Profile successfully retrieved' })
-  @ApiResponse({ status: 401, description: 'Unauthorized' })
   @UseGuards(JwtAuthGuard)
   @Get('me')
   getProfile(@Req() req: any) {
     return req.user;
   }
-} 
\ No newline at end of file
+
+  @ApiOperation({ summary: 'Initiate GitHub connection for logged-in user' })
+  @ApiResponse({ status: 302, description: 'Redirects to GitHub' })
+  @UseGuards(JwtAuthGuard)
+  @Get('connect')
+  @Redirect()
+  async initiateGithubConnect(@Req() req: any) {
+    this.logger.log(`Initiating GitHub connection for user ${req.user.id}`);
+
+    const state = require('crypto').randomBytes(16).toString('hex');
+    
+    if (!req.session) {
+      req.session = {};
+    }
+    
+    req.session.githubOAuthState = state;
+    req.session.fusionAuthId = req.user.fusionAuthId;
+
+    const clientId = this.configService.get('GITHUB_CLIENT_ID');
+    const redirectUri = this.configService.get('GITHUB_REDIRECT_URI');
+    const url = `https://github.com/login/oauth/authorize?client_id=${clientId}&redirect_uri=${encodeURIComponent(redirectUri)}&state=${state}&scope=repo`;
+
+    return { url };
+  }
```

Comment:
ideally this should not be required here since we are using FA and we don't intend to sell a wrapper over it.